### PR TITLE
WT-5246 First pass at incremental backup documentation

### PR DESCRIPTION
**Pull request by sueloverso**

### CODE REVIEW

File: src/docs/backup.dox [hunk 1]
```diff
@@ -89,7 +89,74 @@ rm -rf /path/database.backup &&
...
+Call this \c backup_cursor. Like a normal full backup cursor,
+this cursor will return the filename as the key.  There is no associated
+value.  The information returned will be based on blocks tracked since the time of
+the previous backup designated with "ID1".  New block tracking will be started as
+"ID2" as well.  WiredTiger will maintain modifications from two IDs, the current
+and the most recent completed one.
+
+3. For each file returned by \c backup_cursor->next(), open a duplicate
+backup cursor to do the incremental backup on that file.  The list
+returned will also include log files, \c WiredTigerLog, that need to
```
Comment (ddanderson) [inline src/docs/backup.dox]: I'd change: `log files, \c WiredTigerLog` to `log files (prefixed by \c WiredTigerLog),`

File: src/docs/backup.dox [hunk 2]
```diff
@@ -89,7 +89,74 @@ rm -rf /path/database.backup &&
...
+be copied. Configure that duplicate cursor with \c incremental=(file=name).
+The name comes from the string returned from \c backup_cursor->get_key().
```
Comment (ddanderson) [inline src/docs/backup.dox]: Change `name` to `\c name`

File: src/docs/backup.dox [hunk 3]
```diff
@@ -89,7 +89,74 @@ rm -rf /path/database.backup &&
...
+Call this incr_cursor.
+
+4. The key format for the duplicate backup cursor, \c incr_cursor, is
+\c qqq, representing a file offset and size pair plus a type indicator
+for the range given. There is no associated value. The type indicator
+will be one of WT_BACKUP_FILE or WT_BACKUP_RANGE. For WT_BACKUP_RANGE,
```
Comment (ddanderson) [inline src/docs/backup.dox]: Prefix these `WT_BACKUP...` constants with `\c` throughout this text.

File: src/docs/backup.dox [hunk 4]
```diff
@@ -89,7 +89,74 @@ rm -rf /path/database.backup &&
...
+for each offset/size returned by \c incr_cursor->next(), read the
```
Comment (ddanderson) [inline src/docs/backup.dox]: I think `for each` is confusing here, it implies another nested loop.  I'd say: `For WT_BACKUP_RANGE, read the block indicated by the file offset and size pair from the source database file and write that block to the same offset and size in the backup database file`.

Comment (ddanderson) [reply src/docs/backup.dox]: For my edification, I assume that WT is blocked from writing to the database file until the backup file cursor is closed?

Comment (sueloverso) [reply src/docs/backup.dox]: Not quite. No checkpoints can be deleted, so we're guaranteed that existing blocks will continue to exist. But checkpoint may create new checkpoints that backup is not aware of and that information will not be in the offset/len pairs.

**Comment by sueloverso** [comment]
evergreen retry

**Comment by sueloverso** [comment]
Thanks. I fixed all the suggestions.

**Pull request merged**